library/bigip_user.py

```diff
@@ -182,411 +172,366 @@
-            return True
-        else:
-            return False
+        try:
+            if state == "present":
+                changed = self.present()
+            elif state == "absent":
+                changed = self.absent()
+        except iControlUnexpectedHTTPError as e:
+            raise F5ModuleError(str(e))
+
+        changes = self.changes.to_return()
+        result.update(**changes)
+        result.update(dict(changed=changed))
+        return result
 
-    def user_exists(self):
-        if self.is_version_less_than_13():
-            return self.does_user_exist_without_partition()
+    def present(self):
+        if self.exists():
+            return self.update()
         else:
-            return self.does_user_exist_with_partition()
+            return self.create()
 
-    def does_user_exist_with_partition(self):
-        return self.api.tm.auth.users.user.exists(
-            name=self.params['username_credential'],
-            partition=self.params['partition']
-        )
+    def absent(self):
+        if self.exists():
+            return self.remove()
+        return False
 
-    def does_user_exist_without_partition(self):
-        return self.api.tm.auth.users.user.exists(
-            name=self.params['username_credential']
-        )
+    def should_update(self):
+        result = self._update_changed_options()
+        if result:
+            return True
+        return False
 
-    def update_user(self):
-        params = self.get_changed_parameters()
-        if params:
-            self.changed_params = camel_dict_to_snake_dict(params)
-            if self.params['check_mode']:
-                return True
-        else:
-            return False
-        params['name'] = self.params['username_credential']
-        params['partition'] = self.params['partition']
-        self.update_user_on_device(params)
-        return True
+    def validate_create_parameters(self):
+        """Password credentials and partition access are mandatory,
 
-    def update_user_on_device(self, params):
-        tx = self.api.tm.transactions.transaction
-        with TransactionContextManager(tx) as api:
-            if self.is_version_less_than_13():
-                r = api.tm.auth.users.user.load(
-                    name=self.params['username_credential']
-                )
-                r.modify(**params)
+        when creating a user resource.
+        """
+        if self.want.password_credential is None:
+            err = "The 'password_credential' option " \
+                  "is required when creating a resource."
+            raise F5ModuleError(err)
+        if self.want.partition_access is None:
+            err = "The 'partition_access' option " \
+                  "is required when creating a resource."
+            raise F5ModuleError(err)
+
+    def validate_shell_parameter(self):
+        """Method to validate shell parameters.
+
+        Raise when 'none' is being set for shell parameter,
```

Comment:
why not? so how do you inform the user what was wrong? this behavior is specific to bigip user. shell can be 'none' but we won't get that from device. unless you expect the users to know the intricacies of bigip operation